Modules/arraymodule.c

```diff
@@ -2124,20 +2124,19 @@ array_array___reduce_ex__(arrayobject *self, PyObject *value)
     PyObject *array_str;
     int typecode = self->ob_descr->typecode;
     int mformat_code;
-    static PyObject *array_reconstructor = NULL;
+    _Py_STATICVAR(array_reconstructor);
     long protocol;
     _Py_IDENTIFIER(_array_reconstructor);
     _Py_IDENTIFIER(__dict__);
 
-    if (array_reconstructor == NULL) {
+    if (array_reconstructor.obj == NULL) {
```

Comment:
`_PY_STATICVAR_INIT()` can be used if extract the code for getting `array._array_reconstructor` into a separate function.